File: native/ios_sdk/YogaRenderSDK.swift
import UIKit

public class YogaRenderSDK {
    // Singleton instance
    public static let shared = YogaRenderSDK()
    
    private init() {}
    
    /**
     * Renders content in the provided container view
     * - Parameters:
     *   - containerView: The UIView to render content in
     *   - backgroundColor: Hexadecimal color string (e.g. "#3388FF")
     *   - content: Text content to display
     * - Returns: Boolean indicating success
     */
    public func renderContent(in containerView: UIView, backgroundColor: String, content: String) -> Bool {
        // Configure container view
        containerView.backgroundColor = hexStringToUIColor(hex: backgroundColor)
        
        // Remove any existing content
        containerView.subviews.forEach { $0.removeFromSuperview() }
        
        // Create and configure text label
        let textLabel = UILabel(frame: containerView.bounds)
        textLabel.textAlignment = .center
        textLabel.numberOfLines = 0
        textLabel.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        textLabel.text = "Native SDK Render: \(content)"
        
        // Add label to container
        containerView.addSubview(textLabel)
        
        print("YogaRenderSDK iOS: Content rendered - backgroundColor: \(backgroundColor), content: \(content)")
        return true
    }
    
    /**
     * Creates a UIView with rendered content
     * - Parameters:
     *   - frame: Frame for the view
     *   - backgroundColor: Hexadecimal color string (e.g. "#3388FF")
     *   - content: Text content to display
     * - Returns: UIView with rendered content
     */
    public func createContentView(frame: CGRect, backgroundColor: String, content: String) -> UIView {
        let containerView = UIView(frame: frame)
        _ = renderContent(in: containerView, backgroundColor: backgroundColor, content: content)
        return containerView
    }
    
    // Helper: Convert hex string to UIColor
    private func hexStringToUIColor(hex: String) -> UIColor {
        var hexSanitized = hex.trimmingCharacters(in: .whitespacesAndNewlines)
        hexSanitized = hexSanitized.replacingOccurrences(of: "#", with: "")
        
        // Default to white
        if hexSanitized.isEmpty {
            return UIColor.white
        }
        
        var rgb: UInt64 = 0
        
        if !Scanner(string: hexSanitized).scanHexInt64(&rgb) {
            return UIColor.white
        }
        
        let red = CGFloat((rgb & 0xFF0000) >> 16) / 255.0
        let green = CGFloat((rgb & 0x00FF00) >> 8) / 255.0
        let blue = CGFloat(rgb & 0x0000FF) / 255.0
        
        return UIColor(red: red, green: green, blue: blue, alpha: 1.0)
    }
} 

File: native/android_sdk/src/main/java/com/example/yogarendersdk/YogaRenderSDK.kt
package com.example.yogarendersdk

import android.content.Context
import android.graphics.Color
import android.util.Log
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.TextView

/**
 * YogaRenderSDK - SDK for rendering content in native Android views
 */
class YogaRenderSDK private constructor() {
    companion object {
        private const val TAG = "YogaRenderSDK"
        
        // Singleton instance
        @Volatile
        private var instance: YogaRenderSDK? = null
        
        fun getInstance(): YogaRenderSDK {
            return instance ?: synchronized(this) {
                instance ?: YogaRenderSDK().also { instance = it }
            }
        }
    }
    
    /**
     * Renders content in the provided container view
     * @param containerView The view to render content in
     * @param backgroundColor Hexadecimal color string (e.g. "#3388FF")
     * @param content Text content to display
     * @return Boolean indicating success
     */
    fun renderContent(containerView: ViewGroup, backgroundColor: String, content: String): Boolean {
        try {
            // Set container background color
            containerView.setBackgroundColor(parseColor(backgroundColor))
            
            // Remove any existing content
            containerView.removeAllViews()
            
            // Create and configure text view
            val textView = TextView(containerView.context)
            textView.layoutParams = FrameLayout.LayoutParams(
                FrameLayout.LayoutParams.MATCH_PARENT,
                FrameLayout.LayoutParams.MATCH_PARENT
            )
            textView.textAlignment = TextView.TEXT_ALIGNMENT_CENTER
            textView.gravity = Gravity.CENTER
            textView.text = "Native Render: $content"
            
            // Add to container
            containerView.addView(textView)
            
            Log.d(TAG, "Android: Content rendered - backgroundColor: $backgroundColor, content: $content")
            return true
        } catch (e: Exception) {
            Log.e(TAG, "Error rendering content", e)
            return false
        }
    }
    
    /**
     * Creates a view with rendered content
     * @param context Android context
     * @param backgroundColor Hexadecimal color string (e.g. "#3388FF")
     * @param content Text content to display
     * @return View with rendered content
     */
    fun createContentView(context: Context, backgroundColor: String, content: String): View {
        val containerView = FrameLayout(context)
        containerView.layoutParams = ViewGroup.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        
        renderContent(containerView, backgroundColor, content)
        return containerView
    }
    
    /**
     * Safely parse color string
     * @param colorString Hexadecimal color string
     * @return Parsed color as integer
     */
    private fun parseColor(colorString: String): Int {
        return try {
            Color.parseColor(colorString)
        } catch (e: Exception) {
            Log.w(TAG, "Color parsing failed: $colorString, using default white")
            Color.WHITE
        }
    }
} 